extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/BridgeMDNSDiscoveryParticipant.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.digitalstrom.internal.discovery;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.eclipse.smarthome.binding.digitalstrom.DigitalSTROMBindingConstants;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.config.Config;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.serverConnection.DsAPI;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.serverConnection.impl.DsAPIImpl;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryParticipant;
+import org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryService;
+
+/**
+ * The {@link BridgeMDNSDiscoveryParticipant} is responsible for discovering new and
+ * removed digitalSTROM bridges. It uses the central {@link MDNSDiscoveryService}.
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Matthias Siegele - Initial contribution
+ *
+ */
+public class BridgeMDNSDiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    // private org.slf4j.Logger logger = LoggerFactory.getLogger(DssBridgeMDNSDiscoveryParticipant.class);
+
+    private String hostAdress = null;
+    private String dsid = null;
```

Comment:
neither hostAddress nor dsid should be instance members, because they would be shared across different threads. You could potentially have createResult called two times in parallel for different ServiceInfos.